### ISSUE TITLE
Make "Translator" a role on the website, and use it to manage whether someone is a verified translator or not

### ISSUE THREAD

**Issue by danieljames-dj** [thewca/worldcubeassociation.org#6313]
Currently, we are storing the list of registered translators in two places:
1. https://github.com/thewca/worldcubeassociation.org/wiki/Translating-the-website#existing-translations
2. https://github.com/thewca/worldcubeassociation.org/blob/1298ee9c6f03cf648a7a80ffa26d7661913f9555/WcaOnRails/app/controllers/translations_controller.rb#L53

I've two suggestions regarding this:
1. Can we store the list of translators data in the database? (the only drawback would be that, to add anybody into translators team, we need to add them to the database via SQL query)
2. Can we have a page (like `/translations`) where it will list down all the translators?

**Comment by gregorbg** [comment]
Ideally, I believe we should leverage the existing "Teams and Committees" structure to introduce an (informal) "Translators Team" or something.

However, the current implementation won't allow us to distinguish which member of the Translation Team handles which language. And I am definitely opposed to creating 20+ teams for the sake of translation management.

We could expand the teams structure to add "flags" to team members, like the locale they're translating. This feature could also be backported to the "wst-admin" team to scrap the separate hidden team and simply add "admin" tags to relevant WST members. Thoughts?

**Comment by danieljames-dj** [comment]
I agree with the "flags" idea, it looks good. 👍

**Comment by viroulep** [comment]
Instead of constantly expanding the "team_member" table it may be worth to look into some basic permission table, something where you could have a row looking like this:
```
permission_kind, user_id, team_id
---
update_locale_fr, 277, nil
admin_website, nil, wst
manage_posts, nil, wct
```
Stuff like that, with a very simple UI to list/add/remove permissions (and a permission to manage permission :p).

Pretty sure this is a common issue to multiple CMS and other platforms, so there may be some literature about common architectures to handle this.
It sounds definitely overkill for a simple "locale permission" feature, but I think the usual "extend teams, add more fields, add more methods" approach that we've been using is not the greatest as we have more and more edge cases or hierarchy to implement; and starting testing a basic permission system with something which is not too sensible seems like a good idea.

**Comment by gregorbg** [comment]
My first instinct is that this would be trading one problem for another. Sure, there are a lot of permission systems out there (and even literature, as you suggested) but then how do we go about deciding which permissions are derived through team membership and which permissions are derived through the users table?

Or to phrase it differently, how "granular" should that table be? Won't discussions about the granularity of these permissions lead us into the same dead end as the current "extend teams, add more fields, add more methods" approach, just shifted towards a different table? (I do recognise the current dead end by itself!)

**Comment by denthebro** [comment]
I like @gregorbg's suggestion. This could eliminate those existing team divisions too (like in the WCT currently). Seems elegant to me.

**Comment by saranshgrover** [comment]
I don't think both issues are mutually exclusive - there is #5523 which mentions sub teams to distinguish workflows/sub-teams/gsuite etc where the team and sub team have the same permissions. Then there is the wst-admin subteam, which is made primarily for permissions. I think tagging will eventually lead to permissions anyways - how many tags can you add? etc. A more granular permission based system will give us flexibility for future permissions we might come across.

**Comment by maxidragon** [comment]
This issue can be closed, as we now have translators user_group in the database

**Comment by dunkOnIT** [comment]
We aren't yet using it to manage verified translators as far as I know, though - and we could probably also consider displaying Translator badges? @danieljames-dj thoughts/insights?

**Comment by maxidragon** [comment]
AFAIK, we have it as well - see https://www.worldcubeassociation.org/persons/2013ROGA02

**Comment by danieljames-dj** [comment]
Yes we do manage verified translators with roles ([code link](https://github.com/thewca/worldcubeassociation.org/blob/c9c9de0d00eff3251b3e1dbdfbdb444d77785eed/app/controllers/translations_controller.rb#L61)). We also have the badges.

Thanks for pointing this, @maxidragon. Will close this issue.
